trans.py

```diff
@@ -16,80 +35,85 @@ def find_header(counter, width):
     return -1, -1
 
 def read_pack(name_dfr):
-    '''To read the input info from the Data Format Record'''
+    '''To read the input info from the Data Format Record (DFR)'''
     with open("infos.txt", "r", encoding="UTF-8") as info_read:
         raw_file = info_read.readlines()
         read_data = []
+        types = [[]  for x in range(len(name_dfr))]
+        byte_int = [[] for x in range(len(name_dfr))]
+        names = [[] for x in range(len(name_dfr))]
         for line in raw_file:
             read_data.append(line.strip())  # transform the .txt into a str list w/o '\n'
-        in_pos = read_data.index(name_dfr)  # reference name input  of the Data Format Record group
-        fin_pos = read_data.index("exit"+name_dfr)  # reference to exit the DFR group
-        group_data = []
-        for counter in range(in_pos+1,fin_pos): # to get whole designated DFR pack info from .txt
-            group_data.append(read_data[counter])
-        ref = [0,0,0]
-        ref[0] = group_data.index("type") # ref. mark for type list in .txt
-        ref[1] = group_data.index("bytepack") # mark for byte packs list in .txt
-        ref[2] = group_data.index("varnam") # mark for variable name list in .txt
-        types = []
-        byte_con = []
-        names = []
-        for counter in range(ref[0] + 1, ref[1]):  # to get type of data list
-            types.append(group_data[counter])
-        for counter in range(ref[1] + 1, ref[2]):  # to get byte group length list
-            byte_con.append(group_data[counter])
-        for counter in range(ref[2] + 1, len(group_data)):  # to get var. names list
-            names.append(group_data[counter])
-        byte_int = [int(x) for x in byte_con]
-        return types, byte_int, names   #output in str lists
+        for _x, dat_names in enumerate(name_dfr): #iter. through each filename
+            byte_con = []
+            group_data = []
+            ref = [0,0,0,0,0]
+            ref[0] = read_data.index(dat_names.split('_',1)[0]) # reference to enter DFR group
```

Comment:
If you have to write comments to explain what you're doing, the code is not clear enough. You should only use comments to explain **why** you're doing something.